src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -175,39 +176,142 @@ private BoundPattern BindPropertyPattern(PropertyPatternSyntax node, BoundExpres
         private ImmutableArray<BoundSubPropertyPattern> BindSubPropertyPatterns(PropertyPatternSyntax node, TypeSymbol type, DiagnosticBag diagnostics)
         {
             var result = ArrayBuilder<BoundSubPropertyPattern>.GetInstance();
-            foreach (var syntax in node.PatternList.SubPatterns)
+            foreach (var e in node.Expressions)
             {
-                var propName = syntax.Left;
-                BoundPattern pattern;
-                HashSet<DiagnosticInfo> useSiteDiagnostics = null;
-                LookupResultKind resultKind;
-                Symbol property = FindPropertyOrFieldByName(type, propName, out resultKind, ref useSiteDiagnostics);
-                if ((object)property != null)
+                var syntax = e as IsPatternExpressionSyntax;
```

Comment:
Using an ```as``` cast for a conversion that is expected to always succeed is a violation of our coding guidelines.